RSPeekaboo/Classes/RSPeekaboo.swift

```diff
@@ -8,95 +8,77 @@ import UIKit
 
 
 public class RSPeekaboo {
-    
+
     public enum RSPeekabooState {
         case hidden
         case animating
         case shown
     }
-    
+
     public enum RSPeekabooError: Error {
         case alreadyShowingAPeekaboo
         case showingNone
-        case cannotGetRootView
+        case cannotGetKeyWindow
     }
-    
+
     public static let shared : RSPeekaboo = RSPeekaboo()
     private var mPeekabooState: RSPeekabooState = .hidden
     private var mLastShownView: UIView? = nil
-    
+
     private init() {
-        
+
     }
 
     public func peek(view: UIView, height: CGFloat) throws {
-        guard mPeekabooState == .hidden else {
+        guard self.mPeekabooState == .hidden else {
             throw RSPeekabooError.alreadyShowingAPeekaboo
         }
 
-        guard let rootView = getRootViewControllerView() else {
-            throw RSPeekabooError.cannotGetRootView
+        guard let window = getKeyWindow() else {
+            throw RSPeekabooError.cannotGetKeyWindow
         }
-        
-        // log
-        mPeekabooState = .animating
-        mLastShownView = view
-        
-        let initialRootViewFrame = rootView.frame
-        let finalRootViewSize = CGSize(width: initialRootViewFrame.width, height: initialRootViewFrame.height - height)
-        let finalRootViewFrame = CGRect(origin: initialRootViewFrame.origin, size: finalRootViewSize)
-        
-        
-        view.frame = CGRect(x: 0, y: initialRootViewFrame.maxY, width: initialRootViewFrame.width, height: height)
-        let finalViewFrame = CGRect(x: 0, y: initialRootViewFrame.maxY - height, width: initialRootViewFrame.width, height: height)
-        
-        // added as subview
-        rootView.superview?.addSubview(view)
-
-        rootView.setNeedsLayout()
-        
-        UIView.animate(withDuration: 0.5, animations: { // Allow customizing times
-            rootView.frame = finalRootViewFrame
-            view.frame = finalViewFrame
-            rootView.layoutIfNeeded()
-        }) { (finished) in
-            // Use weak reference
-            self.mPeekabooState = .shown
+
+        self.mPeekabooState = .animating
+        self.mLastShownView = view
```

Comment:
Hi @ratulSharker vai, referring to your comment [here](https://github.com/ratulSharker/RSPeekaboo/pull/7#issuecomment-610254346),

I have kept a reference of the passed view in `peek` method here. This is like your previous implementation. I have not altered anything here. This `mLastShownView` is then used in `hide` method to remove this view from the window.

Reply:
Don't know why i skipped this. My bad 
Btw good work (Y)